pageserver/src/tenant/timeline.rs

```diff
@@ -1312,8 +1369,18 @@ impl Timeline {
             .initial_logical_size
             .get()
             .is_none());
+
+        info!(
+            "spawning logical size computation from context of task kind {:?}",
+            ctx.task_kind()
+        );
```

Comment:
Only now ran into this, was this intentional? I could remove it on #3431..

Reply:
It was kind of intentional, because logical size computation is triggered from multiple places, and will likely be a cause of unwanted layer downloads.

Reply:
Just occured to me that this could be solved 100% with spans. I'll demo this with #3431.